arangod/Aql/ExecutionNode.cpp

```diff
@@ -2582,6 +2584,51 @@ ReturnNode::ReturnNode(ExecutionPlan* plan,
       _inVariable(Variable::varFromVPack(plan->getAst(), base, "inVariable")),
       _count(VelocyPackHelper::getBooleanValue(base, "count", false)) {}
 
+optimizer2::nodes::ReturnNode ReturnNode::toInspectable() const {
+  VPackBuilder becauseWeCan;
+  {
+    if (_inVariable->type() == Variable::Type::Const) {
+      VPackObjectBuilder b(&becauseWeCan);
+    }
+  }
+
+  optimizer2::nodes::ReturnNode planReturnNode = {
+      .inVariable = {
+          .id = _inVariable->id,
+          .name = _inVariable->name,
+          .isFullDocumentFromCollection =
+              _inVariable->isFullDocumentFromCollection,
+          .isDataFromCollection = _inVariable->isFullDocumentFromCollection,
+          .constantValue = becauseWeCan.slice().isObject()
+                               ? std::optional<VPackBuilder>{becauseWeCan}
+                               : std::optional<VPackBuilder>{std::nullopt}}};
+
+  // TODO - fixme start
+  // This whole section will be necessary for every PlanNode which inherits from
+  // BaseNode which is basically every PlanNode ...
+  // This is the "ExecutionNode -> BaseNode (PlanNode)" part.
+  // Reason right now to do it like this is because, we cannot set
+  // embeddedFields directly. I do not like this approach. I want to add some
+  // mechanism that does this for us automatically and not manually like this.
+  // We need to think about a better implementation here.
+  planReturnNode.count = _count;
+  planReturnNode.id = AttributeTypes::Numeric{id().id()};
+  for (auto const& it : _dependencies) {
+    planReturnNode.dependencies.emplace_back(it->id().id());
+  }
+
+  // Think about getting rid of this in total.
+  CostEstimate estimate = getCost();
+  // Think about getting rid of this in total.
+  planReturnNode.estimatedCost = estimate.estimatedCost;
+  // Think about getting rid of this in total.
+  planReturnNode.estimatedNrItems = estimate.estimatedNrItems;
+  // "canThrow" attribute seems to be ignored in this case here.
+  // TODO - fixme end
```

Comment:
@markuspf important todo. We need to solve this "cleanly". Currently just implemented as it is as a reminder. This affects all "PlanNodes".